couchbase/tests/test_couchbase.py

```diff
@@ -193,8 +197,26 @@ def _assert_bucket_metrics(aggregator, tags, device=None):
 
 def _assert_stats(aggregator, node_tags, device=None):
     for mname in NODE_STATS:
-        aggregator.assert_metric('couchbase.by_node.{}'.format(mname), tags=node_tags, count=1, device=device)
+        # These 2 metrics are available ~30 seconds post start. The integrations test won't see it, but the e2e will
+        if mname == 'index_data_size' or mname == 'index_disk_size':
+            aggregator.assert_metric('couchbase.by_node.{}'.format(mname), tags=node_tags, device=device, at_least=0)
+        else:
+            aggregator.assert_metric('couchbase.by_node.{}'.format(mname), tags=node_tags, count=1, device=device)
 
     # Assert 'couchbase.' metrics
     for mname in TOTAL_STATS:
         aggregator.assert_metric('couchbase.{}'.format(mname), tags=CHECK_TAGS, count=1)
+
+
+@pytest.mark.skipif(COUCHBASE_MAJOR_VERSION != 7, reason='Index metrics are only available for Couchbase 7+')
+@pytest.mark.integration
+@pytest.mark.usefixtures("dd_environment")
+def test_index_stats_metrics(aggregator, dd_run_check, instance_index_stats, couchbase_container_ip):
+    """
+    Test Index Statistics metrics (prefixed "couchbase.index." and "couchbase.indexer.")
+    """
+    couchbase = Couchbase('couchbase', {}, [instance_index_stats])
+    dd_run_check(couchbase)
+    for mname in INDEX_STATS_METRICS:
+        aggregator.assert_metric(mname)
+    aggregator.assert_service_check(INDEX_STATS_SERVICE_CHECK_NAME, status=Couchbase.OK, tags=CHECK_TAGS)
```

Comment:
```suggestion
    aggregator.assert_service_check(INDEX_STATS_SERVICE_CHECK_NAME, status=Couchbase.OK, tags=CHECK_TAGS)
    aggregator.assert_metrics_using_metadata(get_metadata_metrics(), check_submission_type=True)
```
Can you also assert that all the metrics are in the `metadata.csv`?